pkg/cloud/aws/services/userdata/controlplane.go

```diff
@@ -282,14 +259,58 @@ func (cpi *ContolPlaneJoinInput) validateCertificates() error {
 }
 
 // NewControlPlane returns the user data string to be used on a controlplane instance.
-func NewControlPlane(input *ControlPlaneInput) (string, error) {
+func NewControlPlane(input *ControlPlaneInput, initConfiguration v1beta1.InitConfiguration) (string, error) {
+	// Override critical variables
+	// TODO(chuckha) add a warning if this is overwriting user input defined
+	// in the configuration.
+	initConfiguration.NodeRegistration.Name = "{{ ds.meta_data.hostname }}"
+	initConfiguration.NodeRegistration.CRISocket = "/var/run/containerd/containerd.sock"
+
+	if initConfiguration.NodeRegistration.KubeletExtraArgs == nil {
+		initConfiguration.NodeRegistration.KubeletExtraArgs = map[string]string{}
+	}
+	initConfiguration.NodeRegistration.KubeletExtraArgs["cloud-provider"] = "aws"
+
+	clusterConfiguration := v1beta1.ClusterConfiguration{
+		APIServer: v1beta1.APIServer{
+			CertSANs: []string{
+				"{{ ds.meta_data.local_ipv4 }}",
+				input.ELBAddress,
+			},
+			ControlPlaneComponent: v1beta1.ControlPlaneComponent{
+				ExtraArgs: map[string]string{
+					"cloud-provider": "aws",
+				},
+			},
+		},
+		ControlPlaneEndpoint: fmt.Sprintf("%s:%d", input.ELBAddress, 6443),
+		ClusterName:          input.ClusterName,
+		Networking: v1beta1.Networking{
+			DNSDomain:     input.ServiceDomain,
+			PodSubnet:     input.PodSubnet,
+			ServiceSubnet: input.ServiceSubnet,
+		},
+		KubernetesVersion: input.KubernetesVersion,
+	}
+	initcfg, err := util.MarshalToYamlForCodecs(&initConfiguration, v1beta1.SchemeGroupVersion, v1alpha1.KubeadmCodecs)
```

Comment:
How do you think we should handle version skew for deploying the different versions of kubeadm? Should we extend the cloud-init module to try to do the conversion through kubeadm (would only support upgrade), or should we leverage the apimachinery tooling here to upgrade/downgrade based on supported version(s)?